dandiapi/api/services/asset/__init__.py

```diff
@@ -140,59 +170,39 @@ def add_asset_to_version(  # noqa: PLR0913, C901
     if zarr_archive and zarr_archive.dandiset != version.dandiset:
         raise ZarrArchiveBelongsToDifferentDandisetError
 
-    # Creating an asset in an OPEN dandiset that points to an embargoed blob results in that
-    # blob being unembargoed
-    unembargo_asset_blob = (
-        asset_blob is not None
-        and asset_blob.embargoed
-        and version.dandiset.embargo_status == Dandiset.EmbargoStatus.OPEN
-    )
     with transaction.atomic():
-        if asset_blob and unembargo_asset_blob:
+        # Creating an asset in an OPEN dandiset that points to an embargoed blob results in that
+        # blob being unembargoed
+        if (
+            asset_blob is not None
+            and asset_blob.embargoed
+            and version.dandiset.embargo_status == Dandiset.EmbargoStatus.OPEN
+        ):
             asset_blob.embargoed = False
             asset_blob.save()
+            transaction.on_commit(
+                lambda: remove_asset_blob_embargoed_tag_task.delay(blob_id=asset_blob.blob_id)
+            )
```

Comment:
```suggestion
            remove_asset_blob_embargoed_tag_task.delay_on_commit(blob_id=asset_blob.blob_id)
```

Reply:
Since this requires ensuring we're on celery `>=5.4`, and we have several other places in the codebase where this could be leveraged, I'd rather do this in a separate PR.

Reply:
https://github.com/dandi/dandi-archive/issues/2006